Networking/Sources/Networking/Networking.swift

```diff
@@ -1,6 +1,61 @@
+import Foundation
+import Alamofire
 public struct Networking {
-    public private(set) var text = "Hello, World!"
 
     public init() {
     }
+    
+    public func perform<M: Decodable, T: TargetType>(target: T, responseClass: M.Type, completion: @escaping (Result<M, Error>) -> ()){
+        let method = target.method.httpMethod
+        let headers = Alamofire.HTTPHeaders(target.headers ?? [:])
+        let params = buildParams(task: target.task)
+        AF.request(target.baseURL + target.path, method: method, parameters: params.0,encoding: params.1, headers: headers).response { response in
+            switch response.result {
+            case .failure(let error):
+                completion(.failure(NetworkingError.networkError(error)))
+            case .success(_):
+                guard let jsonResponse = try? response.result.get() else {
```

Comment:
This strip might not be needed since you get the result value in the `success` case of Alamofire Result